--- a/mt_osal_cfg80211.c
+++ b/mt_osal_cfg80211.c
@@ -2482,6 +2482,9 @@ static int mt_cfg80211_ops_wiphy_params_set(
 
 static int mt_cfg80211_ops_wiphy_params_set(
 	struct wiphy *pWiphy,
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 12, 0))
+	int radio_idx,
+#endif
 	u32 Changed)
 {
 	struct mt_cfg80211_vndr_ctrl ctrl_data, *ctrl_data_ptr = &ctrl_data;
@@ -3224,6 +3227,9 @@ static int mt_cfg80211_ops_set_antenna(
 
 static int mt_cfg80211_ops_set_antenna(
 	struct wiphy *wiphy,
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 12, 0))
+	int radio_idx,
+#endif
 	u32 tx_ant,
 	u32 rx_ant)
 {
@@ -3268,6 +3234,9 @@ static int mt_cfg80211_ops_get_antenna(
 
 static int mt_cfg80211_ops_get_antenna(
 	struct wiphy *wiphy,
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 12, 0))
+	int radio_idx,
+#endif
 	u32 *tx_ant,
 	u32 *rx_ant)
 {
@@ -3494,6 +3503,10 @@ static int mt_cfg80211_ops_get_tx_power(
 static int mt_cfg80211_ops_get_tx_power(
 		struct wiphy *wiphy,
 		struct wireless_dev *wdev,
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 12, 0))
+		int radio_idx,
+		unsigned int link_id,
+#endif
 		int *dbm)
 {
 	struct net_device *dev = NULL;
