--- a/mt_wifi/feature/hostapd_wds/hostapd_wds.c
+++ b/mt_wifi/feature/hostapd_wds/hostapd_wds.c
@@ -28,6 +28,10 @@
 #ifdef HOSTAPD_WDS
 #include "rt_config.h"
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 12, 0))
+#define UNKNOWN_LINK_ID -1
+#endif
+
 INT vlan_wds_virtual_if_open(PNET_DEV pDev)
 {
 	VOID *pAd;
@@ -293,7 +297,11 @@ VOID wds_forward_nulldata(IN
 	MTWF_DBG(pAd, DBG_CAT_AP, CATAP_HOSTAPD_WDS, DBG_LVL_NOTICE,
 				" null data frame from %02x:%02x:%02x:%02x:%02x:%02x, wdev->if_dev[%s]\n",
 				PRINT_MAC(Addr), wdev->if_dev->name);
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 12, 0))
+	if (!cfg80211_rx_unexpected_4addr_frame(wdev->if_dev, Addr, UNKNOWN_LINK_ID, GFP_ATOMIC))
+#else
 	if (!cfg80211_rx_unexpected_4addr_frame(wdev->if_dev, Addr, GFP_ATOMIC))
+#endif
 		MTWF_DBG(NULL, DBG_CAT_AP, CATAP_HOSTAPD_WDS, DBG_LVL_ERROR,
 				"Forward 4addr null data fail\n");
 }
