Index: mt_wifi7/mt_wifi/common/fsm/sta_mgmt_sync.c
===================================================================
--- mt_wifi7.orig/mt_wifi/common/fsm/sta_mgmt_sync.c
+++ mt_wifi7/mt_wifi/common/fsm/sta_mgmt_sync.c
@@ -551,8 +551,8 @@ static BOOLEAN sta_rx_peer_response_upda
 				}
 
 				HW_SET_PROTECT(pAd, wdev, PROT_PROTOCOL, 0, 0);
-				MTWF_DBG(NULL, DBG_CAT_CLIENT, CATCLIENT_APCLI, DBG_LVL_WARN,
-					"SYNC - AP changed B/G protection to %d\n", bUseBGProtection);
+				// MTWF_DBG(NULL, DBG_CAT_CLIENT, CATCLIENT_APCLI, DBG_LVL_WARN,
+				// 	"SYNC - AP changed B/G protection to %d\n", bUseBGProtection);
 			}
 
 #ifdef DOT11_N_SUPPORT
