Index: mt_wifi7/mt_wifi/common/scan.c
===================================================================
--- mt_wifi7/mt_wifi/common/scan.c
+++ mt_wifi7/mt_wifi/common/scan.c
@@ -48,6 +48,7 @@
 
 	PartialScan = (PARTIAL_SCAN *) FunctionContext;
 	PartialScan->bScanning = TRUE;
+	PartialScan->bscanwait = FALSE;
 }
 
 DECLARE_TIMER_FUNCTION(restart_partial_scan);
@@ -1008,6 +1009,7 @@
 			ScanCtrl->Channel = 0;
 			scan_ch_restore(pAd, OpMode, wdev);
 			ScanCtrl->PartialScan.bScanning = FALSE;
+			ScanCtrl->PartialScan.bscanwait = FALSE;
 			if (ScanCtrl->opAbortScan == TRUE)
 				ScanCtrl->opAbortScan = FALSE;
 
@@ -1203,10 +1205,12 @@
 			if (ScanType == SCAN_PARTIAL) {
 				if (ScanCtrl->PartialScan.TimerInterval > 0) {
 					ScanCtrl->PartialScan.bScanning = FALSE;
+					ScanCtrl->PartialScan.bscanwait = FALSE;
 					RTMPSetTimer(&ScanCtrl->PartialScan.PartialScanTimer,
 									ScanCtrl->PartialScan.TimerInterval);
 				} else {
 					ScanCtrl->PartialScan.bScanning = FALSE;
+					ScanCtrl->PartialScan.bscanwait = FALSE;
 					ScanCtrl->PartialScan.pwdev = NULL;
 					ScanCtrl->ScanType = SCAN_ACTIVE;
 				}
@@ -1607,7 +1611,8 @@
 					  ScanInfo->LastScanChannel,
 					  ScanCtrl->PartialScan.bScanning,
 					  ScanCtrl->PartialScan.BreakTime, scan_in_run_state(pAd, wdev));
-			if (scan_in_run_state(pAd, wdev) == FALSE)
+			if ((scan_in_run_state(pAd, wdev) == FALSE)
+				&& (ScanCtrl->PartialScan.bscanwait == FALSE))
 				ScanCtrl->PartialScan.BreakTime++;
 			/* When partialscan is pending, count breaktime;
 			*  We should wait (BeaconPeriod + 100ms), then trigger next scan.*/
@@ -1645,10 +1650,12 @@
 				}
 				NdisZeroMemory(&ScanReq, sizeof(MLME_SCAN_REQ_STRUCT));
 				ScanParmFill(pAd, &ScanReq, ScanCtrl->Ssid, ScanCtrl->SsidLen, BSS_ANY, SCAN_PARTIAL);
+				ScanCtrl->PartialScan.bscanwait = TRUE;
 				if (MlmeEnqueueWithWdev(pAd, SYNC_FSM, SYNC_FSM_SCAN_REQ,
 						sizeof(MLME_SCAN_REQ_STRUCT), &ScanReq, 0, wdev, FALSE, NULL))
 					RTMP_MLME_HANDLER(pAd);
 				else {
+					ScanCtrl->PartialScan.bscanwait = FALSE;
 					ScanCtrl->PartialScan.bScanning = FALSE;
 					if (GetCurrentChannelOpOwner(pAd, wdev) == CH_OP_OWNER_PARTIAL_SCAN)
 						ReleaseChannelOpCharge(pAd, wdev, CH_OP_OWNER_PARTIAL_SCAN);
@@ -1667,6 +1674,7 @@
 
 	PartialScanCtrl = &pAd->ScanCtrl.PartialScan;
 	PartialScanCtrl->bScanning = FALSE;
+	PartialScanCtrl->bscanwait = FALSE;
 	PartialScanCtrl->NumOfChannels = DEFLAUT_PARTIAL_SCAN_CH_NUM;
 	PartialScanCtrl->LastScanChannel = 0;
 	PartialScanCtrl->BreakTime = 0;
Index: mt_wifi7/mt_wifi/common/fsm/fsm_sync.c
===================================================================
--- mt_wifi7/mt_wifi/common/fsm/fsm_sync.c
+++ mt_wifi7/mt_wifi/common/fsm/fsm_sync.c
@@ -1506,6 +1506,7 @@
 		ScanInfo->LastScanChannel, ScanCtrl->BandIdx);
 	/*ASSERT(wdev);*/
 
+	ScanCtrl->PartialScan.bscanwait = FALSE;
 	if (!wdev)
 		return;
 #ifdef CONFIG_6G_SUPPORT

Index: mt_wifi7/mt_wifi/common/fsm/fsm_cntl.c
===================================================================
--- mt_wifi7/mt_wifi/common/fsm/fsm_cntl.c
+++ mt_wifi7/mt_wifi/common/fsm/fsm_cntl.c
@@ -618,6 +618,7 @@
 
 	ScanCtrl->Channel = 0;
 	ScanInfo->LastScanChannel = 0;
+	ScanCtrl->PartialScan.bscanwait = FALSE;
 #ifdef OFFCHANNEL_SCAN_FEATURE
 	ScanCtrl->OffChScan_Ongoing = FALSE;
 	ScanCtrl->state = OFFCHANNEL_SCAN_INVALID;
Index: mt_wifi7/mt_wifi/include/rtmp.h
===================================================================
--- mt_wifi7/mt_wifi/include/rtmp.h
+++ mt_wifi7/mt_wifi/include/rtmp.h
@@ -5404,6 +5404,7 @@
 #define DEFLAUT_PARTIAL_SCAN_BREAK_TIME	4  /* Period of partial scaning: unit: 100ms */
 typedef struct _PARTIAL_SCAN_ {
 	BOOLEAN bScanning;			/* Doing partial scan or not */
+	BOOLEAN	bscanwait;
 	UINT8	 NumOfChannels;			/* How many channels to scan each time */
 	UINT8	 LastScanChannel;		/* last scaned channel */
 	UINT32	 BreakTime;			/* Period of partial scanning: unit: 100ms */
