--- a/mt_wifi/os/linux/cfg80211/cfg80211drv.c
+++ b/mt_wifi/os/linux/cfg80211/cfg80211drv.c
@@ -1811,7 +1811,7 @@ BOOLEAN CFG80211DRV_StaGet(
 	STA_ADMIN_CONFIG *pstacfg = NULL;
 	MAC_TABLE_ENTRY *pEntry = NULL;
 	STA_TR_ENTRY *tr_entry;
-	UINT8 i;
+	UINT8 i, u1Snr = 0; /* SNR From the hardware report, the kernel has no relevant fields, and new reservations are added. */
 	RSSI_SAMPLE rx_rssi_sample;
 	UINT8 RxPath;
 #ifdef VOW_SUPPORT
@@ -1878,6 +1878,26 @@ BOOLEAN CFG80211DRV_StaGet(
 	/* fill tx/rx rate */
 	CFG80211DRV_GetTxRxRate(pAd, pEntry, &pIbssInfo->txrate, &pIbssInfo->rxrate);
 
+	if (pIbssInfo->txrate.legacy) {
+		UINT32 phy_rate    = pIbssInfo->txrate.legacy;  /* Mbps */
+		UINT32 tx_packets  = pIbssInfo->tx_packets;
+		UINT32 tx_failed   = pIbssInfo->tx_failed;
+		UINT32 total       = tx_packets + tx_failed;
+		UINT32 success     = 100;
+		UINT32 thr;
+
+		if (total > 0)
+			success = (tx_packets * 100) / total;
+
+		thr = (phy_rate * 1000) * success * 70 / 100 / 100;
+
+		pIbssInfo->expected_throughput = thr;
+
+		MTWF_DBG(pAd, DBG_CAT_CFG, CATCFG_CMD, DBG_LVL_INFO,
+			"Throughput: phy_rate=%u Mbps, success=%u%%, throughput=%u kbps\n",
+			phy_rate, success, pIbssInfo->expected_throughput);
+	}
+
	/* fill signal */
	pIbssInfo->Signal = RTMPAvgRssi(pAd, &pEntry->RssiSample);
 
@@ -1904,6 +1924,21 @@ BOOLEAN CFG80211DRV_StaGet(
 		pIbssInfo->ack_signal = RTMPAvgRssi(pAd, &rx_rssi_sample);
 	}
 
+	/* fill signal_avg */
+	INT Rssi_temp = 0;
+	RxPath = pAd->Antenna.field.RxPath;
+	RxPath = min(RxPath, (UINT8)(sizeof(rx_rssi_sample.AvgRssi) / sizeof(CHAR)));
+	rx_rssi_sample = pEntry->RssiSample;
+	for (i = 0; i < RxPath; i++) {
+		Rssi_temp += rx_rssi_sample.AvgRssi[i];
+	}
+	pIbssInfo->signal_avg = (CHAR)(Rssi_temp/RxPath);
+
+	/* Fill in SNR per entry*/
+	UniCmdPerStaGetSNR(pAd, pEntry->wcid, &u1Snr);
+	/* fill noise */
+	pIbssInfo->noise = RTMPAvgRssi(pAd, &pEntry->RssiSample) - u1Snr;
+
 	/* fill associated at */
 	pIbssInfo->assoc_at = pEntry->assoc_at;
 
@@ -3152,6 +3187,12 @@ static int CFG80211DRV_OpsStartAp(
 	/* sync ML Probe.rsp Per-STA Profile buffer */
 	bss_mngr_mld_sync_ml_probe_rsp(wdev);
 #endif/*DOT11_EHT_BE*/
+	/*enable getting NF, read CR per 20ms, 2 counts, about 40ms*/
+#ifdef WIFI_UNIFIED_COMMAND
+	UniCmdSetNoiseFloorControl(pAd, UNI_CMD_NF_INFO, 1, 20, 2, 0);
+#else
+	EnableNF(pAd, 1, 20, 2, 0);
+#endif
 end:
 #endif /* CONFIG_AP_SUPPORT */
 	return ret;
