--- a/mt_wifi/os/linux/cfg80211/cfg80211drv.c
+++ b/mt_wifi/os/linux/cfg80211/cfg80211drv.c
@@ -1078,9 +1078,9 @@ void CFG8021DRV_AP_PHYINIT(RTMP
 
 	/*phy mode setting */
 	if (channel > 14) {
-		wdev->PhyMode = (WMODE_A | WMODE_AN | WMODE_AC | WMODE_AX_5G);
+		wdev->PhyMode = (WMODE_A | WMODE_AN | WMODE_AC | WMODE_AX_5G | WMODE_BE_5G);
 	} else {
-		wdev->PhyMode = (WMODE_B | WMODE_G | WMODE_GN | WMODE_AX_24G);
+		wdev->PhyMode = (WMODE_B | WMODE_G | WMODE_GN | WMODE_AX_24G | WMODE_BE_24G);
 	}
 	/*change channel state to NONE*/
 	pchctrl = hc_get_channel_ctrl(pAd->hdev_ctrl);
