--- a/mt_wifi/os/linux/cfg80211/cfg80211_inf.c
+++ b/mt_wifi/os/linux/cfg80211/cfg80211_inf.c
@@ -601,7 +601,7 @@ BOOLEAN RTMP_CFG80211_VirtualIF_Init
 			break;
 		}
 		new_dev_p = os_net_dev_create(MC_RowID, &IoctlIF, inf_type,
-				preIfIndex, sizeof(struct mt_net_dev_priv), preIfName, TRUE);
+				preIfIndex, sizeof(struct mt_net_dev_priv), preIfName, FALSE);
 		if (new_dev_p == NULL) {
 			/* allocation fail, exit */
 			MTWF_DBG(pAd, DBG_CAT_CFG80211, CATCFG80211_INTF, DBG_LVL_ERROR,
