--- a/mt_wifi/protocol/ax/bss_ops.c
+++ b/mt_wifi/protocol/ax/bss_ops.c
@@ -60,6 +60,7 @@ UCHAR ZeroSsid[MAX_LEN_OF_SSID]
 	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
 };
 
+#if 0
 static VOID BssAdjVHTCentCh(BSS_ENTRY *pBss, VHT_OP_IE *vht_op)
 {
 	BOOLEAN hasTwoNonContiguousChannels = (vht_op->vht_op_info.ch_width == VHT_BW_8080);
@@ -81,6 +82,7 @@ static VOID BssAdjVHTCentCh(BSS
 			pBss->SecCentralChannel = vht_op->vht_op_info.ccfs_1;
 	}
 }
+#endif
 
 VOID BssCipherParse(BSS_ENTRY *pBss)
 {
@@ -762,7 +764,7 @@ VOID BssEntrySet
 	pBss->ExtRateLen = rate->ext_rate_len;
 	pBss->Erp  = ie_list->Erp;
 	pBss->Channel = ie_list->Channel;
-	pBss->CentralChannel = ie_list->Channel;
+	//pBss->CentralChannel = ie_list->Channel;
 	pBss->Rssi = Rssi;
 	/* Update CkipFlag. if not exists, the value is 0x0*/
 	pBss->CkipFlag = ie_list->CkipFlag;
@@ -825,8 +855,10 @@ VOID BssEntrySet
 			if (vht_op->vht_op_info.ch_width != VHT_BW_2040 &&
 				cmm_ies->ht_op.AddHtInfo.ExtChanOffset != EXTCHA_NONE &&
 				cmm_ies->ht_cap.HtCapInfo.ChannelWidth == BW_40 &&
-				pBss->CentralChannel != cmm_ies->ht_op.ControlChan)
-				BssAdjVHTCentCh(pBss, vht_op);
+				pBss->CentralChannel != cmm_ies->ht_op.ControlChan) {
+				pBss->CentralChannel = vht_op->vht_op_info.ccfs_0;
+				pBss->SecCentralChannel = vht_op->vht_op_info.ccfs_1;
+			}
 		}
 
 #endif /* DOT11_VHT_AC */
@@ -853,16 +857,18 @@ VOID BssEntrySet
 		SET_HE_OP_EXIST(pBss->ie_exists);
 
 		/*parsing of 6GHz operation ie*/
-		if (pBss->he_ops.he_op_param.param2 & DOT11AX_OP_6G_OPINFO_PRESENT)
+		if (pBss->he_ops.he_op_param.param2 & DOT11AX_OP_6G_OPINFO_PRESENT) {
 			NdisMoveMemory(&pBss->he6g_opinfo, &cmm_ies->he6g_opinfo, sizeof(struct he_6g_op_info));
+			pBss->CentralChannel = cmm_ies->he6g_opinfo.ccfs_0;
+			pBss->SecCentralChannel = cmm_ies->he6g_opinfo.ccfs_1;
+		}
-
 	}
 #ifdef CONFIG_6G_SUPPORT
 	if (HAS_HE_6G_CAP_EXIST(cmm_ies->ie_exists)) {
 		SET_HE_6G_CAP_EXIST(pBss->ie_exists);
 
 		if (cmm_ies->he6g_opinfo.ccfs_0 != 0 && cmm_ies->he6g_opinfo.ccfs_1 != 0)
-			pBss->CentralChannel = cmm_ies->he6g_opinfo.ccfs_1;
+			pBss->SecCentralChannel = cmm_ies->he6g_opinfo.ccfs_1;
 		else if (cmm_ies->he6g_opinfo.ccfs_0 != 0)
 			pBss->CentralChannel = cmm_ies->he6g_opinfo.ccfs_0;
 	}
@@ -882,9 +888,9 @@ VOID BssEntrySet
 	if (HAS_EHT_OP_EXIST(cmm_ies->ie_exists)) {
 		NdisMoveMemory(&pBss->eht_op, &cmm_ies->eht_op, sizeof(struct eht_op_ie));
 		SET_EHT_OP_EXIST(pBss->ie_exists);
+		pBss->CentralChannel = cmm_ies->eht_op.op_info.ccfs0;
+		pBss->SecCentralChannel = cmm_ies->eht_op.op_info.ccfs1;
 
-		if (cmm_ies->eht_op.op_info.control == EHT_BW_320)
-			pBss->CentralChannel = cmm_ies->eht_op.op_info.ccfs1;
 #ifdef CONFIG_MAP_SUPPORT
 		if (PD_CEHCK_MLO_V1_DISABLE(pAd->physical_dev)
 		    || PD_CEHCK_MLO_V1_1_ENABLE(pAd->physical_dev)) {
