--- a/mt_wifi/feature/map/map.c
+++ b/mt_wifi/feature/map/map.c
@@ -483,6 +483,12 @@ INT map_send_bh_sta_wps_done_event
 #if (defined(MTK_MLO_MAP_SUPPORT) && defined(DOT11_EHT_BE))
 			wdev = mac_entry->wdev;
 			entry = get_bss_entry_by_netdev(wdev->if_dev);
+			if (!mac_entry->mlo.mlo_en || !BMGR_VALID_MLO_BSS_ENTRY(entry)) {
+				if (send_event == TRUE && is_ap == 0)
+					bsta_info->setup_link_num = 1;
+				goto out;
+			}
+
 			if (BMGR_VALID_MLO_BSS_ENTRY(entry)) {
 				if (!BMGR_VALID_MLO_DEV(entry->mld_ptr)) {
 					MTWF_DBG(pAd, DBG_CAT_PROTO, CATPROTO_MAP, DBG_LVL_WARN,
