--- a/mt_wifi/common/cmm_sanity.c
+++ b/mt_wifi/common/cmm_sanity.c
@@ -1542,6 +1542,7 @@ BOOLEAN PeerBeaconAndProbeRspSanity
 				}
 			}
 #endif /*DOT11_EHT_BE*/
+			copy_to_vie((UCHAR *)pVIE, LengthVIE, ptr_eid, pEid);
 			break;
 #ifdef DOT11V_MBSSID_SUPPORT
 
@@ -1566,6 +1567,7 @@ BOOLEAN PeerBeaconAndProbeRspSanity
 				"cache rnr ie\n");
 			NdisMoveMemory(ie_list->cmm_ies.rnr_ie, pEid, pEid->Len + 2);
 			SET_RNR_EXIST(cmm_ies->ie_exists);
+			copy_to_vie((UCHAR *)pVIE, LengthVIE, ptr_eid, pEid);
 			break;
 #endif
 		default:
--- a/mt_wifi/os/linux/cfg80211/cfg80211drv.c
+++ b/mt_wifi/os/linux/cfg80211/cfg80211drv.c
@@ -2847,7 +2847,7 @@ BOOLEAN CFG80211_checkScanTable
 			CenFreq = ieee80211_channel_to_frequency(pBssEntry->Channel, IEEE80211_BAND_2GHZ);
 
 		chan = ieee80211_get_channel(pWiphy, CenFreq);
-		ieLen = 2 + pApCliEntry->MlmeAux.SsidLen + pBssEntry->VarIeFromProbeRspLen;
+		ieLen = 2 + pApCliEntry->MlmeAux.SsidLen + pBssEntry->VarIELen;
 		os_alloc_mem(NULL, (UCHAR **)&ie, ieLen);
 
 		if (!ie) {
@@ -2859,8 +2859,8 @@ BOOLEAN CFG80211_checkScanTable
 		ie[0] = WLAN_EID_SSID;
 		ie[1] = pApCliEntry->MlmeAux.SsidLen;
 		NdisCopyMemory(ie + 2, pApCliEntry->MlmeAux.Ssid, pApCliEntry->MlmeAux.SsidLen);
-		NdisCopyMemory(ie + 2 + pApCliEntry->MlmeAux.SsidLen, pBssEntry->pVarIeFromProbRsp,
-					   pBssEntry->VarIeFromProbeRspLen);
+		NdisCopyMemory(ie + 2 + pApCliEntry->MlmeAux.SsidLen, pBssEntry->VarIEs,
+				pBssEntry->VarIELen);
 		bss = cfg80211_inform_bss(pWiphy, chan,
 								  pApCliEntry->MlmeAux.Bssid, timestamp, WLAN_CAPABILITY_ESS, pApCliEntry->MlmeAux.BeaconPeriod,
 								  ie, ieLen,
