Index: hostapd-2022-07-29-b704dc72/src/utils/config.c
===================================================================
--- hostapd-2022-07-29-b704dc72.orig/src/utils/config.c
+++ hostapd-2022-07-29-b704dc72/src/utils/config.c
@@ -68,6 +68,10 @@
 		 */
 		sstart = pos;
 		end = os_strchr(sstart, '#');
+		if (os_memcmp(sstart, "ssid=", 5) == 0 || os_memcmp(sstart, "psk=", 4) == 0 || os_memcmp(sstart, "sae_password=", 13) == 0)
+		{
+			end = NULL;
+		}
 		while (end) {
 			sstart = os_strchr(sstart, '"');
 			if (!sstart || sstart > end)
