--- a/wpa_supplicant/Makefile
+++ b/wpa_supplicant/Makefile
@@ -1986,7 +1986,7 @@
 OBJS_priv += ../src/drivers/driver_common.o
 
 OBJS += wpa_supplicant.o events.o bssid_ignore.o wpas_glue.o scan.o
-OBJS_t := $(OBJS) $(OBJS_l2) eapol_test.o
+OBJS_t := $(OBJS) $(OBJS_l2) eapol_test.o $(OBJS_d)
 OBJS_t += ../src/radius/radius_client.o
 OBJS_t += ../src/radius/radius.o
 OBJS_t2 := $(OBJS) $(OBJS_l2) preauth_test.o
