--- a/iwinfo_nl80211.c
+++ b/iwinfo_nl80211.c
@@ -403,11 +403,15 @@ static int nl80211_phy_idx_from_uci(const
 
 static int nl80211_phy_idx_from_uci(const char *name)
 {
-	struct uci_section *s;
+	struct uci_section *s = NULL;
 	const char *opt;
 	int idx = -1;
-
-	s = iwinfo_uci_get_radio(name, "mac80211");
+	const char *drv_types[] = {"mtwifi", "mtk", "mac80211"};
+	const char **type;
+
+	for (type = drv_types; *type && !s; type++) {
+		s = iwinfo_uci_get_radio(name, *type);
+	}
 	if (!s)
 		goto out;
 
